LPD-68187 users should also be visible from suborganizations

wanderlast opened the pull request:
Hi @smottal ,

https://liferay.atlassian.net/browse/LPD-68187, tied to LPP.

Fix is for an issue where users who manage suborganizations cannot see the members of that suborganization when searching users. I've set it so that the logic demands both manage users AND manage suborganizations permission for visibility, but if this should be its own separate permission (e.g. "manage suborganizations users") or should just be tied to "manage suborganizations" please let me know and I'll adjust the solution from there.

An integration test has also been included. Thanks!

liferay-continuous-integration commented:
Test suite sf has been triggered on http://test-1-33

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-33.liferay.com/job/test-portal-source-format/4450">test-portal-source-format#4450</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-33.liferay.com/userContent/jobs/test-portal-source-format/builds/4450/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-commerce/liferay-portal/pull/6774">liferay-commerce#6774</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/342184435">[master] ci:test:sf - wanderlast &gt; liferay-commerce - PR#6774 - 2025-10-17[16:53:40]</a><br/></div><div><strong>Testray Build ID:</strong><span></span><code>342184435</code><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-33.liferay.com/job/test-portal-source-format/4450/">test-portal-source-format#4450</a><br/></div></div>

wanderlast commented:
@smottal sure, I'll ask on Slack. And currently we cannot see the users of the suborganization at all even with the 'manage suborganization' permission. With the manage users permission we can see the users of the 'parent' org with or without the 'manage suborganization' permission.

wanderlast commented:
Hi @smottal ,

I've updated the PR due to the discussion on Slack. I've added a new permission "Manage Suborganizations Users" and have used it to determine visibility of suborganization users regardless of MANAGE_USERS. I've modified the test to take into account the updated behavior as well.

smottal commented on modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/internal/search/spi/model/permission/contributor/test/UserSearchPermissionFilterContributorTest.java:
testWhenHasOrganizationManageSuborganizations**Users**PermissionSearch

wanderlast commented:
ci:test:sf

smottal commented:
ci:test:relevant

smottal commented:
ci:report:352327231

liferay-continuous-integration commented:
Test suite default has been triggered on http://test-1-32

liferay-continuous-integration commented:
<h1>Build completed.</h1><p>Testray CSV has been generated successfully for testrayBuildID: <b>352327231</b>.</p><p>Job Link: <a href="https://test-1-32.liferay.com/job/generate-testray-csv/889/">generate-testray-csv</a></p><p>Testray CSV Link: <a href="https://test-1-32.liferay.com/userContent/jobs/generate-testray-csv/builds/889/testray-results-352327231.csv">testray-results-352327231.csv</a></p>

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/167175
[Console](https://test-1-35.liferay.com/job/forward-pullrequest/787//consoleFull)
